src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/collectors/ShardIndexingPressureMetricsCollector.java

```diff
@@ -0,0 +1,210 @@
+
+public class ShardIndexingPressureMetricsCollector extends PerformanceAnalyzerMetricsCollector implements MetricsProcessor {
+    public static final int SAMPLING_TIME_INTERVAL = MetricsConfiguration
+        .CONFIG_MAP.get(ShardIndexingPressureMetricsCollector.class).samplingInterval;
+    private static final int KEYS_PATH_LENGTH = 0;
+    private static final Logger LOG = LogManager.getLogger(ShardIndexingPressureMetricsCollector.class);
+    private static final ObjectMapper mapper = new ObjectMapper();
+    private static final JSONParser parser = new JSONParser();
+
+    public static final String SHARD_INDEXING_PRESSURE_CLASS_NAME = "org.elasticsearch.index.ShardIndexingPressure";
+    public static final String CLUSTER_SERVICE_CLASS_NAME = "org.elasticsearch.cluster.service.ClusterService";
+    public static final String INDEXING_PRESSURE_CLASS_NAME = "org.elasticsearch.index.IndexingPressure";
+    public static final String SHARD_INDEXING_PRESSURE_STORE_CLASS_NAME = "org.elasticsearch.index.ShardIndexingPressureStore";
+    public static final String INDEXING_PRESSURE_FIELD_NAME = "indexingPressure";
+    public static final String SHARD_INDEXING_PRESSURE_FIELD_NAME = "shardIndexingPressure";
+    public static final String SHARD_INDEXING_PRESSURE_STORE_FIELD_NAME = "shardIndexingPressureStore";
+    public static final String SHARD_INDEXING_PRESSURE_HOT_STORE_FIELD_NAME = "shardIndexingPressureHotStore";
+
+    private final ConfigOverridesWrapper configOverridesWrapper;
+    private final PerformanceAnalyzerController controller;
+    private StringBuilder value;
+
+    public ShardIndexingPressureMetricsCollector(PerformanceAnalyzerController controller,
+                                                 ConfigOverridesWrapper configOverridesWrapper) {
+        super(SAMPLING_TIME_INTERVAL, "ShardIndexingPressureMetricsCollector");
+        value = new StringBuilder();
+        this.configOverridesWrapper = configOverridesWrapper;
+        this.controller = controller;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void collectMetrics(long startTime) {
+        if(!controller.isCollectorEnabled(configOverridesWrapper, getCollectorName())) {
+            return;
+        }
+
+        try {
+            ClusterService clusterService = ESResources.INSTANCE.getClusterService();
+            if (clusterService != null) {
+                IndexingPressure indexingPressure = (IndexingPressure) getField(CLUSTER_SERVICE_CLASS_NAME, INDEXING_PRESSURE_FIELD_NAME).get(clusterService);
+                if(indexingPressure != null) {
+                    Object shardIndexingPressure = getField(INDEXING_PRESSURE_CLASS_NAME, SHARD_INDEXING_PRESSURE_FIELD_NAME).get(indexingPressure);
+                    Object shardIndexingPressureStore = getField(SHARD_INDEXING_PRESSURE_CLASS_NAME, SHARD_INDEXING_PRESSURE_STORE_FIELD_NAME).get(shardIndexingPressure);
+                    Map<Long, Object> shardIndexingPressureHotStore =
```

Comment:
Did we do stress testing here for shardIndexingPressure metric collection ? What is max size of `shardIndexingPressureHotStore` allowed and how much time did the metric collection take for this max size?